apis/v1alpha1/gateway_types.go

```diff
@@ -281,6 +283,20 @@ type Listener struct {
 	// are compatible with the application protocol specified in
 	// the Protocol field.
 	//
+	// Multiple Routes may match a request to this Listener. If that happens,
+	// precedence MUST be given in order of the following criteria:
+	// * The match with the longest combination of hostname and path.
```

Comment:
This one is HTTP-specific, maybe we should move it to that route?

Reply:
We should replace this with 'More specific matches should be given precedence over less specific ones.'.

Reply:
Good catch, thanks! I wasn't sure how to fit this into HTTPRoute so I added that as an example for a more generic statement like Harry suggested. Maybe in the future it would be good to define specificity per route as well, but right now the only route we have with native matching is HTTPRoute (TCP and UDP both rely on extensionRef).